### PR TITLE
[Merged by Bors] - refactor: move analyticGroupoid to a new file `AnalyticManifold.lean`

### PR DESCRIPTION
**Pull request by girving**
It used to be in `SmoothManifoldWithCorners.lean`, but this makes no sense as nothing downstream of that file uses `analyticGroupoid`. In #10853, I will rewrite the definition of `analyticGroupoid` and define `AnalyticManifold` on top of it.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Geometry/Manifold/AnalyticManifold.lean [hunk 1]
```diff
@@ -0,0 +1,192 @@
+/-
+Copyright (c) 2024 Geoffrey Irving. All rights reserved.
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/AnalyticManifold.lean:2]: If I understand correctly, mathlib practice is to put the original author(s) of the code you're moving: in this case, using git blame to find out who wrote this code. Give a moment...

**Review by grunweg — COMMENTED** [review]
Thank you for following up so quickly! This looks good to me (it's straightforward code motion); I like the new module docstring, a happy by-product is that `SmoothManifoldWithCorners` can shrink a little (that file is quite large anyway).

I just have one comment about the copyright header.

File: Mathlib/Geometry/Manifold/AnalyticManifold.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Geometry/Manifold/AnalyticManifold.lean:2]: Turns out this was written in mathlib4, in #6386. Please update the header accordingly.

Comment (girving) [reply Mathlib/Geometry/Manifold/AnalyticManifold.lean]: Oops, fixed. Apologies: I'd copied those lines from the broader commit, where most of the new file was mine.

**Comment by grunweg** [comment]
Thanks for the swift response! I still think this looks good, so let's
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**